cocos/network/CCDownloader-apple.mm

```diff
@@ -515,7 +515,18 @@ - (void)URLSession:(NSURLSession *)session downloadTask :(NSURLSessionDownloadTa
     NSString *destPath = [NSString stringWithUTF8String:storagePath];
     NSFileManager *fileManager = [NSFileManager defaultManager];
     NSURL *destURL = nil;
-    
+
+    // On iOS 9 a response with status code 4xx(Client Error) or 5xx(Server Error)
+    // might end up calling this delegate method, saving the error message to the storage path
```

Comment:
It is definitely caused by the different behaviour of iOS runtime on iOS 8 and iOS 9. I installed the same app on 4 different devices(iOS 8.3 & iOS 8.4.1 & iOS 9.0 & iOS 9.1), iOS 8.x all returns `You do not have permission to access the requested resource` and failed the download, while iOS 9.x all saved the error response in the storage path(as state in https://github.com/cocos2d/cocos2d-x/issues/14677 ) @pandamicro